processing/src/main/java/io/druid/query/dimension/CascadeExtractionDimensionSpec.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.dimension;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.metamx.common.StringUtils;
+import io.druid.query.extraction.CascadeExtractionFn;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class CascadeExtractionDimensionSpec implements DimensionSpec
+{
+  private static final byte CACHE_TYPE_ID = 0x4;
+
+  private final String dimension;
+  private final CascadeExtractionFn extractionFn;
+  private final String outputName;
+
+  @JsonCreator
+  public CascadeExtractionDimensionSpec(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("outputName") String outputName,
+      @JsonProperty("extractionFns") ExtractionFn[] extractionFns,
+      @JsonProperty("cascadeExtractionFn") CascadeExtractionFn cascadeExtractionFn
```

Comment:
Can you have an "inner" and "outer" function? one concern I have with an array is that the order of application is not intuitive like it is with nested json. You can cascade multiple cascading extraction functions by having the "inner" function set to another cascading extraction function.